gossipd/routing.c

```diff
@@ -35,6 +36,33 @@ struct pending_node_announce {
 	u32 index;
 };
 
+/* We consider a reasonable gossip rate to be 1 per day, with burst of
+ * 4 per day.  So we use a granularity of one hour. */
+#define TOKENS_PER_MSG 24
+#define TOKEN_MAX (24 * 4)
+
+static bool ratelimit(const struct routing_state *rstate,
+		      u8 *tokens, u32 prev_timestamp, u32 new_timestamp)
+{
+	u64 num_tokens;
+
+	assert(new_timestamp >= prev_timestamp);
+
+	/* First, top up tokens, avoiding overflow. */
+	num_tokens = *tokens + ((new_timestamp - prev_timestamp)
+				/ GOSSIP_TOKEN_TIME(rstate->dev_fast_gossip));
```

Comment:
This may lead us to effectively allow fewer than 1 update per day, since the division will round down.

Reply:
Not a big problem in ratelimiting, but it may have some strange effects if other implementations expect to be able to send one a day, we may end up just getting one every two days.

Reply:
In practice, 1 per day is walking the edge already I think.